geode-redis/src/main/java/org/apache/geode/redis/internal/pubsub/Subscription.java

```diff
@@ -53,11 +53,15 @@ void publishMessage(byte[] channel, byte[] message,
   List<Object> createResponse(byte[] channel, byte[] message);
 
   /**
-   * Return the subscription name. In the case of a pattern the string representation of the
-   * pattern is returned.
+   * Return the channel name. In the case of a pattern null is returned.
    */
   byte[] getChannelName();
 
+  /**
+   * Return the pattern name. In the case of a channel null is returned.
+   */
+  byte[] getPatternName();
```

Comment:
I can see that you want the method name to be more expressive, but I don't think it's good practice to have an API where some methods return null depending on what type of object you're using. That opens the door for a dev to easily make a mistake. I'd prefer to keep a single method but give it a more descriptive name. Perhaps `getSubscriptionName()` or `getChannelOrPattern()`.

Reply:
I'm definitely open to suggestions on this, maybe we could have another method that's called `getChannelOrPattern()` or some other way of distinguishing whether a channel/pattern name was returned?

Reply:
I see what the problem is now. I think it would be better to provide a `Type` enum into the `Subscription` interface:
```
  enum Type {
    CHANNEL,
    PATTERN;
  }

  Type getType();
```

And then you should be able to filter by that when you need to have methods distinguish the type of subscription.

In the `PubSub` interface you can add:
```
List<byte[]> findSubscribedChannels(Client client, Subscription.Type type);
```

Reply:
Good idea, I just pushed a change. Let me know what you think!